### DISCUSSION TITLE
Certificate renewal without opening port 80?

### DISCUSSION THREAD

**Discussion by mjmare** [caprover/caprover#1723] [General]
I have run into this situation where I run Caprover on an internal server. Now access from the internet on all external ports, including 80 and 443 is blocked (I had them open when installing Caprover). This prevents Certbot from renewing the certificates. 
This makes a nginx health check fail, which seems to lead to other trouble (eg installing apps fails often, and I have to log in time and time again). 
So my question is: can I run certbot manually to renew the certs periodically? Or can I use another (nonstandard) port perhaps? Or can I use DNS challenge instead of the HTTP challenge? 
Thanks in advance for any suggestion.

**Comment by maietta** [comment]
This is kind of a stupid question, but why would you setup SSL's via LetsEncrypt for something you won't be able to use later?

In any case, you *could* use WireGuard on super cheap VPS as a server, and then WireGuard as a client on the CapRover server and expose your server to the internet, punching a whole in NAT or CNAT, whatever you have that normally blocks inbound ports 80 and 443.

Once you have confirmed networking between sever and client, you can then setup Haproxy to reverse proxy all requests to port 80 and 443 to your new network interface created by WireGuard.

Once that's completed, you'll have access to your CapRover server via the IP of your VPS. You can then point a domain (or domains) to this IP.

This is my setup for a Raspberry Pi running Caprover. I pay $3.50/mo for a VPS with Vultr to proxy my requests to my IP using the IP of the VPS. It works quite nice, even though my Pi is on Starlink or 4G LTE internet.

**Reply by mjmare** [reply]
My caprover instance is available to the internal network, but not reachable from the internet. I thought it would be nice to have HTTPS on the local network. 
I could live with plain HTTP if the caprover/caprover container didn't kill itself every 3 minutes when failing to renew the certs.

**Reply by githubsaturn** [reply]
You can disable health check by skipping domain verification

**Reply by mjmare** [reply]
Thanks, that helped in preventing Caprover from restarting. 
If only I could find a way to manually refresh the certificates on a nonstandard port. Maybe something like:
```
docker exec -it <certbot container> certbot renew --http-01-port=8080
```
Then I could use a script to open the non standard port temporarily. Clumsy but it might work.

**Comment by githubsaturn** [comment]
> So my question is: can I run certbot manually to renew the certs periodically? Or can I use another (nonstandard) port perhaps? Or can I use DNS challenge instead of the HTTP challenge?

The answer to all these questions are no - as mentioned in the docs, CapRover only works with the standard ports:
![image](https://user-images.githubusercontent.com/10582082/231938802-30b40146-ab84-4a85-a824-b0a5dc6d99b9.png)
